app/controllers/spree/admin/search_controller.rb

```diff
@@ -19,7 +19,12 @@ def known_users
 
       def customers
         @customers = []
-        if spree_current_user.enterprises.pluck(:id).include? search_params[:distributor_id].to_i
+        enterprise_ids = if spree_current_user.admin?
+                           Enterprise.pluck(:id)
```

Comment:
Loading all enterprise ids just to check if one specific id is present seemed inefficient to me. I added a couple of commits to simplify this.